src/overlays/actors/ovl_En_Rl/z_en_rl.c

```diff
@@ -9,81 +16,379 @@ void EnRl_Destroy(Actor* thisx, GlobalContext* globalCtx);
+    if (func_80AE74B4(this, globalCtx, 4, 0)) {
+        this->action = 5;
+        this->drawConfig = 2;
+        this->alpha = 0;
+        this->actor.shape.unk_14 = 0;
+        this->unk_19C = 0.0f;
+    }
+}
+
+void func_80AE79A4(EnRl* this, GlobalContext* globalCtx) {
+    f32* unk_19C = &this->unk_19C;
+    s32 alpha = 255;
+
+    if (func_80AE74B4(this, globalCtx, 4, 0)) {
+        *unk_19C += 1.0f;
+        if (*unk_19C >= kREG(5) + 10.0f) {
+            this->action = 7;
+            this->drawConfig = 1;
+            *unk_19C = kREG(5) + 10.0f;
+            this->alpha = alpha;
+            this->actor.shape.unk_14 = alpha;
+            return;
+        }
+    } else {
+        *unk_19C -= 1.0f;
+        if (*unk_19C <= 0.0f) {
+            this->action = 4;
+            this->drawConfig = 0;
+            *unk_19C = 0.0f;
+            this->alpha = 0;
+            this->actor.shape.unk_14 = 0;
+            return;
+        }
+    }
+    alpha = (*unk_19C / (kREG(5) + 10.0f)) * 255.0f;
+    this->alpha = alpha;
+    this->actor.shape.unk_14 = alpha;
+}
+
+void func_80AE7AF8(EnRl* this, GlobalContext* globalCtx) {
+    if (func_80AE74B4(this, globalCtx, 3, 0)) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_0600040C, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_0600040C.genericHeader), 2, -8.0f);
+        this->action = 6;
+    } else if (func_80AE74FC(this, globalCtx, 4, 0)) {
+        this->action = 5;
+        this->drawConfig = 2;
+        this->unk_19C = kREG(5) + 10.0f;
+        this->alpha = 255;
+        if (this->unk_1A4 == 0) {
+            func_80AE78D4(this, globalCtx);
+            this->unk_1A4 = 1;
+        }
+        this->actor.shape.unk_14 = 0xFF;
+    }
+}
+
+void func_80AE7BF8(EnRl* this, s32 arg1) {
+    if (arg1 != 0) {
+        SkelAnime_ChangeAnim(&this->skelAnime, &D_06000830, 1.0f, 0.0f,
+                             SkelAnime_GetFrameCount(&D_06000830.genericHeader), 0, 0.0f);
+        this->action = 7;
+    }
+}
+
+void func_80AE7C64(EnRl* this, GlobalContext* globalCtx) {
+    func_80AE7954(this, globalCtx);
+    func_80AE73D8(this, globalCtx);
+}
+
+void func_80AE7C94(EnRl* this, GlobalContext* globalCtx) {
+    func_80AE744C(this, globalCtx);
+    func_80AE7494(this);
+    func_80AE72D0(this);
+    func_80AE79A4(this, globalCtx);
+    func_80AE73D8(this, globalCtx);
+}
+
+void func_80AE7CE8(EnRl* this, GlobalContext* globalCtx) {
+    s32 temp;
+
+    func_80AE744C(this, globalCtx);
+    temp = func_80AE7494(this);
+    func_80AE72D0(this);
+    func_80AE7BF8(this, temp);
+    func_80AE73D8(this, globalCtx);
+}
+
+void func_80AE7D40(EnRl* this, GlobalContext* globalCtx) {
+    func_80AE744C(this, globalCtx);
+    func_80AE7494(this);
+    func_80AE72D0(this);
+    func_80AE7AF8(this, globalCtx);
+    func_80AE73D8(this, globalCtx);
+}
+
+void func_80AE7D94(EnRl* this, GlobalContext* globalCtx) {
+    s32 pad[2];
+    s16 temp = this->unk_190;
+    s32 addr = D_80AE81A0[temp];
```

Comment:
with `D_80AE81A0` being an array of eye textures, this makes unk_190 an `eyeTextureIndex` or something, feel free to find a better name lol